src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -0,0 +1,148 @@
+            "household_details",
+            source=self.get_household_details,
+            requires_columns=["household_id", "tracked"],
+        )
+
+    ##################################
+    # Pipeline sources and modifiers #
+    ##################################
+
+    def get_household_details(self, idx: pd.Index) -> pd.DataFrame:
+        """Source of the household_details pipeline"""
+        pop = self.population_view.subview(["tracked", "household_id"]).get(idx)
+        household_details = (
+            pop[["household_id"]]
+            .astype(HOUSEHOLD_ID_DTYPE)
+            .join(
+                self._households,
+                on="household_id",
+            )
+        )
+        household_details = household_details.astype(HOUSEHOLD_DETAILS_DTYPES)
+        return household_details
+
+    ##################
+    # Public methods #
+    ##################
+
+    def create_households(
+        self, num_households: int, housing_type: str = "Standard"
+    ) -> pd.Series:
+        """
+        Create a specified number of new households, with new, unique address_ids.
+
+        Parameters
+        ----------
+        num_households
+            The number of households to create.
+        housing_type
+            The housing type of the new households.
+
+        Returns
+        -------
+        household_ids for the new households.
+        The length of this Series will be num_households.
+        """
+        household_ids = self._next_available_ids(num_households, taken=self._households.index)
+        address_ids = self._next_available_ids(
+            num_households, taken=self._households["address_id"]
+        )
+
+        new_households = pd.DataFrame(
+            {
+                "address_id": address_ids,
+                "housing_type": housing_type,
+            },
+            index=household_ids.astype(HOUSEHOLD_ID_DTYPE),
+        ).astype(HOUSEHOLD_DETAILS_DTYPES)
+
+        self._households = pd.concat([self._households, new_households])
+
+        return household_ids
+
+    def update_household_addresses(self, household_ids: pd.Series) -> None:
+        """
+        Changes the address_id associated with each household_id in household_ids
+        to a new, unique value.
+
+        Parameters
+        ----------
+        household_ids
+            The IDs of the households that should move to new addresses.
+        """
+        self._households, _ = update_address_ids(
+            moving_units=self._households,
+            units_that_move_ids=household_ids,
+            starting_address_id=self._first_int_available(self._households["address_id"]),
+            address_id_col_name="address_id",
+        )
+
+    ##################
+    # Helper methods #
+    ##################
+
+    @staticmethod
+    def _next_available_ids(num_ids: int, taken: pd.Series):
+        return Households._first_int_available(taken) + np.arange(num_ids)
```

Comment:
Note: this approach is assuming households are never deleted from the households data structure (they aren't, but we should be clear that we are relying on this).